docs/life_of_a_godwoken_transaction.md

```diff
@@ -0,0 +1,338 @@
+
+![Godwoken Architecture](./images/godwoken.png)
+
+* Individual parties, named *aggregators* can collect layer 2 Godwoken transactions(more details on layer 2 transactions will be explained below), then pack them into layer 2 blocks, and submit them to CKB. Due to the nature of optimistic rollup, aggregators will need to stake a certain amount of CKB in order to submit a layer 2 block.
+* Each layer 2 block submitted to the chain is first marked as **unfinalized**. An unfinalized layer 2 block might be challenged by others(named **challengers**), if the layer 2 block appears malicious.
+   + A challenger starts a challenge request by marking one layer 2 transaction in the layer 2 block as in valid. Some CKB will need to be staked in order to create a challenge. Once a start challenge request is submitted, the challenge phase is started.
+   + In a challenge phase(details will be explained below), the original aggregator submitting the block will need to prove that the marked layer 2 transaction is indeed correct. The aggregator can do this by submitting a cancel challenge request, which executes the marked layer 2 transaction on chain. CKB only accepts valid transaction on chain, which means the layer 2 transaction must succeed for the cancel challenge request to be committed on chain. In a cancel challenge request, aggregator can claim staked tokens from the challenger.
+   + If an aggregator fails to prove the validity of the layer 2 transaction via a cancel challenge request, the challenger can claim tokens staked by the aggregator, and revert the chain to the state just before the invalid layer 2 block.
+* After a certain amount of time, an unfinalized layer 2 block will become finalized. Finalized layer 2 blocks will be freezed, and cannot be challenged by anyone.
+
+Note this is slightly different from a typical optimistic rollup: here a transaction is executed, only to prove the transaction is valid, while a timeout indicates challenge success in this case.
+
+In this design, an aggregator offers liquidity costs for staking CKBs used in submitting layer 2 block, as well as layer 1 transaction fees used in the layer 1 transactions containing layer 2 block. In exchange, the aggregator can charge layer 2 transaction fees from layer 2 users. As explained below, layer 2 transaction fees can be paid either in CKB or in any kind of [sUDT](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md). In fact, Godwoken treat CKB and sUDTs as exactly the same thing in layer 2.
+
+## PoA Deployments
+
+Godwoken is designed based on the assumption that anyone shall be able to issue layer 2 blocks. However, in reality, this might not be feasible for different cases: a public layer 2 blockchain will have additional requirements, that blocks must only be issued at certain intervals to prevent congestions; a private layer 2 blockchain might be only issued by certain parties.
```

Comment:
```suggestion
Godwoken is designed based on the assumption that anyone shall be able to propose layer 2 blocks. However, in reality, this might not be feasible for different cases: a public layer 2 blockchain will have additional requirements, that blocks must only be proposed at certain intervals to prevent congestions; a private layer 2 blockchain might be only proposed by certain parties.
```